### ISSUE TITLE
what is the purpose of wc_set_time_limit(0) ?

### ISSUE THREAD

**Issue by erikdemarco** [woothemes/woocommerce#31080]
In woocommerce its used oftenly. So if for example there is a bug. It will run forever and take cpu resources. Isnt it bad practice to set_time_limit to forever?

**Comment by tammullen** [comment]
Hi @erikdemarco 

While our goal is to address all the issues reported in this repository, GitHub should be treated as a place to report confirmed bugs only.

I am adding the `needs developer feedback` label to this issue so that the WooCommerce Core team can try take a look at your question. Please note it may take a few days for them to get to this issue.

However please note due to the number of bug reports being submitted, we are prioritizing work on them and may not be able to attend to all questions. 
​
The best place to get help further is in the [WooCommerce Community Slack](https://woocommerce.com/community-slack/) in the `#developers` channel.

**Comment by barryhughes** [comment]
I tend to agree. I would think it makes more sense to use a lower value and reapply it as needed (such as in cases where we are in a loop, processing a large number of actions: it could be reset on each iteration).

**Comment by jamesckemp** [comment]
This issue has been automatically closed as part of the backlog cleanup announced in [Refining how we manage the WooCommerce backlog](https://developer.woocommerce.com/2026/05/29/backlog-cleanup/). It was identified as a support request or an issue that belongs outside this repository, rather than a bug report for WooCommerce core. For help with your store, please use the [WooCommerce.com support channels](https://woocommerce.com/support/), or the [community forum](https://wordpress.org/support/plugin/woocommerce/) for the free plugin. If you believe this issue was closed in error, please leave a comment and we will take another look.
